common/dynamic_bitarray.h

```diff
@@ -0,0 +1,82 @@
+/*
+ *  nextpnr -- Next Generation Place and Route
+ *
+ *  Copyright (C) 2021  Symbiflow Authors
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ *  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ *  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ *  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ *  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ *  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ *  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+
+#include <cstdint>
+#include <limits>
+#include <vector>
+
+// This class implements a simple dynamic bitarray, backed by some resizable
+// random access storage.  The default is to use a std::vector<uint8_t>.
+
+#ifndef DYNAMIC_BITARRAY_H
+#define DYNAMIC_BITARRAY_H
+
+namespace nextpnr {
```

Comment:
This should be NEXTPNR_NAMESPACE_BEGIN like in other files

Reply:
I'm avoiding depending on `nextpnr.h` where possible because it causes header nightmares where include order can cause terrible compiler errors.

Reply:
Yep, I can see that. The original idea for `NEXTPNR_NAMESPACE_BEGIN` was to enable multiple nextpnrs to be built with different namespaces and linked together but afaik no-one is using that (it would need a custom main if nothing else) but this has never happened so `namespace nextpnr` is OK for now.